### PR TITLE
Add the ability to receive the hierarchy of files as an Iterable.

### PR DESCRIPTION
**Pull request by dgroup**

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#956](https://github.com/yegor256/cactoos/issues/956) is now in scope, role is `REV`

File: src/main/java/org/cactoos/io/file/ChildrenOf.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    private final UncheckedScalar<Iterable<File>> src;
+
+    /**
+     * Ctor.
+     * @param parent The parent file.
+     */
+    public ChildrenOf(final File parent) {
+        this(
+            () -> new Ternary<List<File>>(
+                () -> parent == null
```
Comment (llorllale) [inline src/main/java/org/cactoos/io/file/ChildrenOf.java]: @dgroup let's skip this null-check - it's the user's responsibility to provide a valid reference

**Comment by llorllale** [comment]
@dgroup we already have `org.cactoos.io.Directory`. Does it not satisfy your requirements?

**Comment by dgroup** [comment]
Hi @llorllale ,
Thanks a lot. The `Directory` is more than enough.

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#956` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#956](https://github.com/yegor256/cactoos/issues/956) is now out of scope

**Pull request closed without merging**